packages/babel-plugin-proposal-private-property-in-object/src/native-private-fields.js

```diff
@@ -0,0 +1,112 @@
+import syntaxPlugin from "@babel/plugin-syntax-private-property-in-object";
+import { injectInitialization as injectConstructorInit } from "@babel/helper-create-class-features-plugin";
+
+export default function pluginPrivateIn({ types: t, template }) {
+  const ids = new WeakMap();
+
+  function unshadow(name, targetScope, scope) {
+    while (scope !== targetScope) {
+      if (scope.hasOwnBinding(name)) scope.rename(name);
+      scope = scope.parent;
+    }
+  }
+
+  function injectInitialization(classPath, init) {
+    let firstFieldPath;
+    let consturctorPath;
+
+    for (const el of classPath.get("body.body")) {
+      if (
+        (el.isClassProperty() || el.isClassPrivateProperty()) &&
+        !el.node.static
+      ) {
+        firstFieldPath = el;
+        break;
+      }
+      if (!consturctorPath && el.isClassMethod({ kind: "constructor" })) {
+        consturctorPath = el;
+      }
+    }
+
+    if (firstFieldPath) {
```

Comment:
When doing `#foo in instance`, `instance` doesn't have the field until `#foo = init` completes. It's not correct to do this before the first field runs (the v8.dev [blog post](https://v8.dev/features/private-brand-checks) just reminded me of this, see the `halfConstructed` example).

Reply:
Nice I hate it

Reply:
Yes, accessors/methods are installed before the first initializer is run, so they cannot fail to install.